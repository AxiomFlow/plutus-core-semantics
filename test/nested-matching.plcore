(module Foo (imported) (exported () ()) (local () ())
  (define result
    (case (con Nat.Pair (con Nat.Pair 1 2) (con Nat.Pair 3 4))
          (cl Nat.NotThis ()  -1)
          (cl Nat.Pair (p1 p2)
            (case p2
              (cl Foo.NotThis (x) -2)
              (cl Nat.Pair (x y) (builtin multiplyInt (x y)))
              (cl Foo.NotThisEither (x) -2))))))
