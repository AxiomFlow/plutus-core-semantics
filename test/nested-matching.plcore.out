<T> <k> . </k> <env> ( Foo . result ) |-> 12 </env> <mod> Foo </mod> </T>
